### PR TITLE
Use str_starts_with/str_ends_with

### PR DESCRIPTION
**Pull request by fiwswe**
Implement the proposed changes from #4060

### CODE REVIEW

File: inc/Utf8/PhpString.php [hunk 1]
```diff
@@ -27,9 +27,8 @@ public static function basename($path, $suffix = '')
             $path = substr($path, $rpos + 1);
         }
 
-        $suflen = strlen($suffix);
-        if ($suflen && (substr($path, -$suflen) === $suffix)) {
-            $path = substr($path, 0, -$suflen);
+        if (str_ends_with($path, $suffix)) {
+            $path = substr($path, 0, -strlen($suffix));
```
Comment (Klap-in) [inline inc/Utf8/PhpString.php:31]: Empty suffix results in str_end_with == true, does that make a difference here?

**Review by Klap-in — CHANGES_REQUESTED** [review]
There are failing unit tests, I cannot yet discover why, but these failures might be introduced by this PR. Could you check the tests?

File: inc/Extension/PluginController.php [hunk 2]
```diff
@@ -351,7 +351,7 @@ protected function getListByType($type, $enabled)
             if (is_dir($typedir)) {
                 if ($dp = opendir($typedir)) {
                     while (false !== ($component = readdir($dp))) {
-                        if (strpos($component, '.') === 0 || strtolower(substr($component, -4)) !== '.php') continue;
+                        if (strpos($component, '.') === 0 || str_ends_with(strtolower($component), '.php')) continue;
```
[the remark this answers is not in the document]
Comment (fiwswe) [reply inc/Extension/PluginController.php]: Yep, fixed. Good catch.

File: inc/init.php [hunk 3]
```diff
@@ -553,7 +553,7 @@ function is_ssl()
  */
 function isWindows()
 {
-    return strtoupper(substr(PHP_OS, 0, 3)) === 'WIN';
+    return (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN');
```
[the remark this answers is not in the document]
Comment (fiwswe) [reply inc/init.php]: Fixed-

File: inc/Utf8/PhpString.php (hunk 1 above)
Comment (fiwswe) [reply inc/Utf8/PhpString.php]: I have looked at this as well. At first glance I don't see any effective difference. But if `$suffix === ''` then the statement `$path = substr($path, 0, -strlen($suffix));` becomes: `$path = substr($path, 0, -0);` and `-0 === 0` so yeah, that will make a difference.
Fixed.

**Comment by fiwswe** [comment]
> There are failing unit tests, I cannot yet discover why, but these failures might be introduced by this PR. Could you check the tests?

Yes, I had the same problem — couldn't figure out where the failures came from. I'm hoping they where caused by the subtle PhpString logic error … but alas no, the failures persist.

File: inc/Utf8/PhpString.php (hunk 1 above)
Comment (Klap-in) [reply inc/Utf8/PhpString.php]: This solved also all issues with the unit tests :-)

**Comment by fiwswe** [comment]
> but alas no, the failures persist.

Actually, you were right. The failures are gone. Apparently I was viewing the logs from an older (overlapping) run.

Ok then, any objections to merging this?

**Comment by fiwswe** [comment]
@Klap-in Thanks, also for your help in finding and fixing the issues :-)

**Pull request merged**